src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,54 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static final int MIN_LOGIN_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new InvalidUserDataException("User shouldn't be null!");
+        }
+        if (checkUserAge(user)
+                && checkUserNotRegisteredYet(user)
+                && checkUserLoginLength(user)
+                && checkUserPasswordLength(user)) {
+            storageDao.add(user);
+        }
```

Comment:
I think we do not need `boolean` return value in validation methods. That causes additional if statement to be placed, which is redundant. Anyway, if validation fails there will be an exception, so feel free to make them void.